include/votca/csg/io/pdbwriter.h

```diff
@@ -0,0 +1,200 @@
+
+namespace votca {
+namespace csg {
+
+/**
+ * @brief
+ *
+ * Note that by default the coordinates stored in the pdb file are in units
+ * of Angstroms while the following repositories have default units of:
+ *
+ * csg: nm
+ * xtp: bohr
+ *
+ * Further note that pdb files store the ids, serical numbers and indices
+ * starting at 1, votca uses ids, indices and serial numbers starting at 0.
+ */
+template <typename Bead_T, class Molecule_T, class Topology_T>
+class PDBWriter : public TrajectoryWriter {
+ public:
+  void RegisteredAt(
+      tools::ObjectFactory<std::string, TrajectoryWriter> &factory) {}
+
+  void Write(boost::any conf);
+
+  template <class T>
+  void WriteContainer(Topology_T *conf, T &container);
+
+  void WriteHeader(std::string header);
+
+  void WriteBox(const Eigen::Matrix3d &box);
+
+ private:
+  template <class T>
+  std::string getType(T &atom) {
+    std::string type = atom.getType();
+    if (type.size() > 4) {
+      type = type.substr(0, 4);
+    }
+    return type;
+  }
+
+  template <class T>
+  std::string getElement(T &item) {
+    if (item.getElement() == tools::topology_constants::unassigned_element) {
+      return "";
+    }
+    return item.getElement();
+  }
+
+  /**
+   * @brief Get the residue type
+   *
+   * In the case that the residue type is unknown it will follow the pdb
+   * convention and assign it a value of "UNK", see the url for more info:
+   *
+   * www.wwpdb.org/documentation/file-format-content/format33/sect4.html
+   *
+   * @tparam Atom
+   * @param atom
+   *
+   * @return
+   */
+  template <class T>
+  std::string getResidueType(T &item) {
+    std::string restype = item.getResidueType();
+    if (restype == tools::topology_constants::unassigned_residue_type) {
+      restype = "UNK";
+    } else if (restype.size() > 3) {
+      restype = restype.substr(0, 3);
+    }
+    return restype;
+  }
+
+  template <class T>
+  int getId(T &item) {
+    return item.getId() + 1;
+  }
+
+  template <class T>
+  int getResId(T &item) {
+    return item.getResidueId() + 1;
+  }
+
+  template <class T>
+  Eigen::Vector3d getPos(T &item) {
+    return item.getPos() * tools::conv::bohr2ang;
+  }
+
+  Eigen::Vector3d getPos(Bead_T &bead) {
+    return bead.Pos() * tools::conv::nm2ang;
+  }
+
+  template <class T>
+  std::vector<Bead_T *> getIterable(Topology_T &top, T &container) {
+    std::vector<Bead_T *> beads;
+    std::vector<int> bead_ids = container.getBeadIds();
+    for (int &bead_id : bead_ids) {
+      beads.push_back(top.getBead(bead_id));
+    }
+    return beads;
+  }
+
+  std::vector<Bead_T *> getIterable(Topology_T &top) {
+    std::vector<Bead_T *> beads;
+    std::vector<int> bead_ids = top.getBeadIds();
+    for (int &bead_id : bead_ids) {
+      beads.push_back(top.getBead(bead_id));
+    }
+    return beads;
+  }
+};
+
+template <class Bead_T, class Molecule_T, class Topology_T>
+template <class T>
+inline void PDBWriter<Bead_T, Molecule_T, Topology_T>::WriteContainer(
```

Comment:
I do not undderstand this method, why do we have so many template parameters here